amaranth/build/plat.py

```diff
@@ -307,8 +309,8 @@ def toolchain_prepare(self, fragment, name, *, ports, emit_src=True, **kwargs):
         # and to incorporate the Amaranth version into generated code.
         autogenerated = f"Automatically generated by Amaranth {__version__}. Do not edit."
 
-        rtlil_text, self._name_map = rtlil.convert_fragment(fragment, ports=ports, name=name,
-                                                            emit_src=emit_src)
+        rtlil_text, self._name_map = rtlil.convert_fragment(fragment, name=name,
+                                                            emit_src=emit_src, propagate_domains=False)
```

Comment:
I think this might be a stale option?

Reply:
Oh, no, it does get passed to `prepare()` down the road doesn't it

Reply:
yup